pkg/feature/manifest.go

```diff
@@ -1,76 +1,103 @@
 package feature
 
 import (
+	"bytes"
+	"embed"
 	"fmt"
 	"html/template"
-	"os"
+	"io"
+	"io/fs"
 	"path/filepath"
 	"strings"
-
-	"github.com/pkg/errors"
 )
 
-const BaseOutputDir = "/tmp/odh-operator"
+//go:embed templates
+var embeddedFiles embed.FS
 
 type manifest struct {
 	name,
-	path string
+	path,
+	processedContent string
 	template,
-	patch,
-	processed bool
+	patch bool
 }
 
-func loadManifestsFrom(path string) ([]manifest, error) {
+func loadManifestsFrom(fsys fs.FS, path string) ([]manifest, error) {
 	var manifests []manifest
-	if err := filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+
+	err := fs.WalkDir(fsys, path, func(path string, dirEntry fs.DirEntry, err error) error {
 		if err != nil {
 			return err
 		}
-		if info.IsDir() {
+
+		if dirEntry.IsDir() {
 			return nil
 		}
-		basePath := filepath.Base(path)
-		manifests = append(manifests, manifest{
-			name:     basePath,
-			path:     path,
-			patch:    strings.Contains(basePath, ".patch"),
-			template: filepath.Ext(path) == ".tmpl",
-		})
+
+		_, err = fs.ReadFile(fsys, path)
+		if err != nil {
+			return err
+		}
+		m := loadManifestFrom(path)
+		manifests = append(manifests, m)
 
 		return nil
-	}); err != nil {
-		return nil, errors.WithStack(err)
+	})
+
+	if err != nil {
+		return nil, err
 	}
 
 	return manifests, nil
 }
 
+func loadManifestFrom(path string) manifest {
```

Comment:
`load` here is a bit misleading as we are not loading anything yet at this point... maybe simply `createManifestFrom` is a bit better.

Reply:
agree, done here [54f4f38](https://github.com/opendatahub-io/opendatahub-operator/pull/797/commits/54f4f3835be46134b21a2ec5dba81e2517bfe93d)